itest/rust/src/registration/constant_test.rs

```diff
@@ -0,0 +1,188 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use crate::itest;
+use godot::{
+    engine::ClassDb,
+    prelude::{meta::ClassName, *},
+};
+
+#[derive(GodotClass)]
+struct HasConstants {}
+
+#[godot_api]
+impl HasConstants {
+    #[constant]
+    const A: i64 = 128;
+
+    #[constant]
+    const B: i128 = -600;
+
+    #[constant]
+    const C: u8 = u8::MAX;
+
+    #[constant]
+    const D: usize = 20 + 33 * 45;
```

Comment:
How does the implementation work, do we just look at the right-hand side value and whether that is losslessly convertible to `i64`?

Maybe we should try to register the constant via name instead, i.e. `HasConstants::A`? That way we make sure the type is considered.

Reply:
It now looks like:
```rs
fn __register_constants() {
    use ::godot::builtin::meta::registration::constant::*;
    use ::godot::builtin::meta::ClassName;
    use ::godot::builtin::StringName;
    ExportConstant::new(
        ClassName::of::<HasConstants>(),
        ConstantKind::Integer(IntegerConstant::new(StringName::from("A"), HasConstants::A)),
    )
    .register();
    ExportConstant::new(
        ClassName::of::<HasConstants>(),
        ConstantKind::Integer(IntegerConstant::new(StringName::from("B"), HasConstants::B)),
    )
    .register();
    ExportConstant::new(
        ClassName::of::<HasConstants>(),
        ConstantKind::Integer(IntegerConstant::new(StringName::from("C"), HasConstants::C)),
    )
    .register();
    ExportConstant::new(
        ClassName::of::<HasConstants>(),
        ConstantKind::Integer(IntegerConstant::new(StringName::from("D"), HasConstants::D)),
    )
    .register();
}
```

Keeping in mind that `IntegerConstant::new` takes a `TryInto<i64>`

Reply:
I see, thanks!

It means however that too big integers will only be detected at startup time. Which I guess is OK if we want to be flexible regarding types -- otherwise we'd need to enforce `i64` (or manually list safe conversions) or require literals.

Reply:
i think this solution is fine as-is. you'll get the warning on startup at least so it's not like this is something that can randomly crash your program. and it's by far the more flexible solution.

If you'd prefer a different solution then sure but i prefer this one out of the ones i can think of.